src/HotChocolate/Core/src/Execution/DependencyInjection/SchemaRequestExecutorBuilderExtensions.Middleware.cs

```diff
@@ -87,36 +77,115 @@ public static partial class SchemaRequestExecutorBuilderExtensions
         return builder.ConfigureSchema(b => b.Use(middleware));
     }
 
-    private sealed class MapMiddleware
+    private static IRequestExecutorBuilder MapFieldMiddleware(
+        this IRequestExecutorBuilder builder,
+        FieldReference fieldReference,
+        FieldMiddleware middleware)
+        => builder
+            .TryAddTypeInterceptor(typeof(ApplyFieldMiddlewareInterceptor))
+            .ConfigureSchema(
+                b => b.SetContextData(ApplyFieldMiddlewareInterceptor.ContextKey,
+                obj =>
+                {
+                    if (obj is not FieldMiddlewareLookup lookup)
+                    {
+                        lookup = new FieldMiddlewareLookup();
+                    }
+
+                    lookup.RegisterFieldMiddleware(fieldReference, middleware);
+                    return lookup;
+                }));
+
+    private sealed class FieldMiddlewareLookup
     {
-        private readonly FieldDelegate _next;
-        private readonly FieldReference _fieldReference;
-        private readonly FieldDelegate _fieldDelegate;
-
-        public MapMiddleware(
-            FieldDelegate next,
-            FieldReference fieldReference,
-            FieldDelegate fieldDelegate)
+        private readonly IDictionary<string, List<FieldMiddlewareReference>> _lookup =
+            new Dictionary<string, List<FieldMiddlewareReference>>();
+
+        public bool HasFieldMiddleware(string typeName) =>
+            _lookup.ContainsKey(typeName);
+
+        public bool TryGetFieldMiddlewares(
+            string type,
+            string field,
+            [NotNullWhen(true)] out IEnumerable<FieldMiddleware>? middleware)
+        {
+            middleware = null;
+            if (!_lookup.TryGetValue(type, out var references))
+            {
+                return false;
+            }
+
+            middleware = references
```

Comment:
I guess linq is oke here, otherwise i need to allocate a list.

Reply:
look at my change... since its all private I think now we save the allocations :) by exposing the list